### ISSUE TITLE
Question: Hideable header and collapsible rows

### ISSUE THREAD

**Issue by jkganzan13** [bencripps/react-redux-grid#35]
Hi Ben,

Is there a way to hide the grid header row? Based on the documentation, I can only find an option to hide the columns but not the header or any rows for that matter.

Also, does the grid support a collapsible row similar to this http://jsfiddle.net/rnqZu/2/? A row click event should open a new row underneath the clicked one.

Many Thanks!

**Comment by bencripps** [comment]
For hiding rows and the header, currently the grid doesn't support such an action. But I'm ok with adding an action to do it.

It would look something like

````js
store.dispatch(
    Actions.GridActions.hideRow({
        stateKey: 'unique',
        rowIndex: 0
    })
);
````

And to hide the header

````js
store.dispatch(
    Actions.GridActions.hideHeader({
        stateKey: 'unique'
    })
);
````

I like the row expander feature, but this is a much larger concept, which would require more time. The first question I have is how does the `grid` receive the `rowExpander` prop, and where does that data occur on the row?

Initially, I think you would install `rowExpander` as a plugin:

````js
const plugins = {
    ROW_EXPANDER: {
        enabled: true,
        isExpandable: ({ row, column, columns, store }) => {
            // this function would determine if the parent row
            // should be expandable
            if (row.values.name === 'Michael') {
                return true;
            }
            return false;
        },
        children: ({ row }) => {
            // inside the row object would be the rowValues
            return <div> This div would be shown when row is expanded </div>
        }
    }
};
````

When I get some time, I will add the `actions` to hide the header and rows, but let me know what you think about the `ROW_EXPANDER` API. If you have any suggestions, we can talk about how it should be consumed.

**Comment by jkganzan13** [comment]
My grid rows should all be expandable and I was thinking of rendering a component within each expanded row, so your `ROW_EXPANDER` implementation is perfect, and more customizable for others. 

I was going to suggest adding a separate datasource for the child rows similar to the remote datasource for the grid, but if I can render a component in the child row, I think I can just pass the parent id as props and dispatch an action in that component. 

Can't wait til you implement these features. Thanks Ben!

**Comment by bencripps** [comment]
@jkganzan13 I've add the header action, which will work when you call the following action:

````js
store.dispatch(
    Actions.GridActions.setHeaderVisibility({
        stateKey: 'unique', hidden: true
    })
);
```

I will publish a new release with these changes.

**Comment by FreeClimb** [comment]
Hi
I use your Grid component as a report on the work of the service organization for the period.
I display the hierarchy of departments, engineers and their orders in the first column. I calculate the data locally when I go to the report page. I need to stop expanding rows by condition. What can be done?
Thanks

**Comment by bencripps** [comment]
The `ROW_EXPANDER` plugin hasn't been implemented yet :/ so this functionality doesn't currently exist. I will mark this as help wanted and get to it as soon as I can.

**Comment by FreeClimb** [comment]
Yes, I understand that this functionality has not yet been implemented
Thank you very much for your prompt reply
I'll wait when it happens

**Comment by bencripps** [comment]
Just curious, does setting expandable on the nodes which do/dont need to be expandable solve this problem for you?

**Comment by FreeClimb** [comment]
Yes, of course, Ben
Now, as a workaround, I added CSS my classes to the employee and order nodes, with the help of ColRenderer. And then on the button onClick handler, I unexpanded tr of table contained span with my class.

**Comment by FreeClimb** [comment]
Ben, one more suggestion:
You pass for formatting in function renderer one parameter object, contained column, row and cell.
However, these parameters do not allow formatting an <td> element, containing a cell. 
It would be convenient to add a ref to <td>element in the parameters renderer function
Now to set backgroundColor of the <td> element, I perform the following:
```
renderer: ({ row }) => {
                        return (
                            <span className="percent-cell" style={{ backgroundColor: row.agree_percent.bgColor }}>
                                {row.agree_percent.str}
                            </span>);
                    }
```
and add code in componentDidMount() and componentDidUpdate()
```
    styleTreeGrid() {
        let cells = document.getElementsByClassName('percent-cell');
        let i;
        for (i = 0; i < cells.length; i++) {
            cells[i].parentNode.parentNode.style.backgroundColor = cells[i].style.backgroundColor;
            cells[i].parentNode.parentNode.style.textAlign = 'center';
        }

        cells = document.getElementsByClassName('text-align-cell');
        for (i = 0; i < cells.length; i++) {
            cells[i].parentNode.parentNode.style.textAlign = 'center';
        }
    }

    componentDidMount() {
        this.styleTreeGrid();
    }

    componentDidUpdate() {
        this.styleTreeGrid();
    }
```
